File: assignments/a2/q2/Python/BitOne.py
#BitOne:top#
#BitOne:import#

# <bit>BitOne ::= <ONE>
class BitOne(Bit): #BitOne:class#

    className = "BitOne"
    ruleString = "<bit>BitOne ::= <ONE>"
    one = None

    def __init__(one):
        #BitOne:init#
        self.one = one

#BitOne#

File: assignments/a2/q2/Java/BitOne.java
//BitOne:top//
//BitOne:import//
import java.util.*;

// <bit>BitOne ::= <ONE>
public class BitOne extends Bit /*BitOne:class*/ {

    public static final String $className = "BitOne";
    public static final String $ruleString =
        "<bit>BitOne ::= <ONE>";

    public Token one;

    public BitOne(Token one) {
//BitOne:init//
        this.one = one;
    }

    public static BitOne parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<bit>BitOne", scn$.lno);
        Token one = scn$.match(Token.Match.ONE, trace$);
        return new BitOne(one);
    }

    public int eval(Bit b) {
        return 1;
    }

//BitOne//
}
